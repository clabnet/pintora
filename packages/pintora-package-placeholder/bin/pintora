#!/usr/bin/env node

const str = `
This is a placeholder package, in case the name 'pintora' is taken by other projects.

Pintora is an extensible text-to-diagrams library that works in both browser and node.js.

Please check out https://pintorajs.vercel.app/docs/getting-started/usage/ for more instructions of using @pintora/cli in Node.js.
`

console.log(str)
