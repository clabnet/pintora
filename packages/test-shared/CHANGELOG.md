# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)


### Features

* **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))





# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)


### Features

* **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))





# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)


### Features

* [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))





# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)


### Bug Fixes

* **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))


### Features

* **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))





# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)


### Features

* Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
