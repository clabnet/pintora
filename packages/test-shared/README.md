# `@pintora/test-shared`

> TODO: description

## Usage

```
const testShared = require('@pintora/test-shared');

// TODO: DEMONSTRATE API
```
