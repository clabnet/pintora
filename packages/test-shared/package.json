{
  "name": "@pintora/test-shared",
  "version": "0.0.0",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  }
}
