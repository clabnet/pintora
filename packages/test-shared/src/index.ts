export { EXAMPLES } from './data/examples'
export { stripStartEmptyLines } from './util'
