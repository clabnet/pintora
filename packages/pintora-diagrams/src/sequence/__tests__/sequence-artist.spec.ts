import pintora from '@pintora/core'
import { EXAMPLES } from '@pintora/test-shared'
import { testDraw, prepareDiagramConfig } from '../../__tests__/test-util'
import { sequenceDiagram } from '../index'

describe('sequence-artist', () => {
  beforeAll(() => {
    prepareDiagramConfig()
    pintora.diagramRegistry.registerDiagram('sequenceDiagram', sequenceDiagram)
  })

  it('match example snapshot', () => {
    expect(testDraw(EXAMPLES.sequence.code).graphicIR).toMatchSnapshot()
  })
})
