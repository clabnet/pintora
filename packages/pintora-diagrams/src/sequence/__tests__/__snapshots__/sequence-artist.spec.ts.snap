// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sequence-artist draw cross mark to 1`] = `
Object {
  "mark": Object {
    "attrs": Object {},
    "children": Array [
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [],
        "class": "activations",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "lineDash": Array [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-0-4",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "Message",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
        ],
        "class": "message",
        "itemId": "message-0-4",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist draw participant classifier 1`] = `
Object {
  "mark": Object {
    "attrs": Object {},
    "children": Array [
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "height": 38,
                    "lineWidth": 1.5,
                    "r": 7.6,
                    "stroke": "#3b4044",
                    "width": 53.199999999999996,
                    "x": 40,
                    "y": 7.6,
                  },
                  "type": "circle",
                },
                Object {
                  "attrs": Object {
                    "fill": null,
                    "lineWidth": 1.5,
                    "path": Array [
                      Array [
                        "M",
                        28.6,
                        21.53333333333333,
                      ],
                      Array [
                        "L",
                        51.4,
                        21.53333333333333,
                      ],
                      Array [
                        "M",
                        40,
                        15.2,
                      ],
                      Array [
                        "L",
                        40,
                        26.6,
                      ],
                      Array [
                        "l",
                        -7.6,
                        15.2,
                      ],
                      Array [
                        "l",
                        7.6,
                        -15.2,
                      ],
                      Array [
                        "l",
                        7.6,
                        15.2,
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "path": Array [
                      Array [
                        "M",
                        103.4,
                        12,
                      ],
                      Array [
                        "L",
                        103.4,
                        26,
                      ],
                      Array [
                        "A",
                        26.599999999999998,
                        12,
                        0,
                        1,
                        0,
                        156.6,
                        26,
                      ],
                      Array [
                        "L",
                        156.6,
                        12,
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
                Object {
                  "attrs": Object {
                    "cx": 130,
                    "cy": 19,
                    "fill": "#fdb05e",
                    "rx": 26.599999999999998,
                    "ry": 12,
                    "stroke": "#3b4044",
                    "x": 130,
                    "y": 12,
                  },
                  "type": "ellipse",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "path": Array [
                      Array [
                        "M",
                        220,
                        19,
                      ],
                      Array [
                        "m",
                        -26.599999999999998,
                        0,
                      ],
                      Array [
                        "l",
                        26.599999999999998,
                        19,
                      ],
                      Array [
                        "l",
                        26.599999999999998,
                        -19,
                      ],
                      Array [
                        "l",
                        -26.599999999999998,
                        -19,
                      ],
                      Array [
                        "Z",
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "C",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-C",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [],
        "class": "activations",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "height": 38,
                    "lineWidth": 1.5,
                    "r": 7.6,
                    "stroke": "#3b4044",
                    "width": 53.199999999999996,
                    "x": 40,
                    "y": 77.6,
                  },
                  "type": "circle",
                },
                Object {
                  "attrs": Object {
                    "fill": null,
                    "lineWidth": 1.5,
                    "path": Array [
                      Array [
                        "M",
                        28.6,
                        91.53333333333333,
                      ],
                      Array [
                        "L",
                        51.4,
                        91.53333333333333,
                      ],
                      Array [
                        "M",
                        40,
                        85.2,
                      ],
                      Array [
                        "L",
                        40,
                        96.6,
                      ],
                      Array [
                        "l",
                        -7.6,
                        15.2,
                      ],
                      Array [
                        "l",
                        7.6,
                        -15.2,
                      ],
                      Array [
                        "l",
                        7.6,
                        15.2,
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "path": Array [
                      Array [
                        "M",
                        103.4,
                        82,
                      ],
                      Array [
                        "L",
                        103.4,
                        96,
                      ],
                      Array [
                        "A",
                        26.599999999999998,
                        12,
                        0,
                        1,
                        0,
                        156.6,
                        96,
                      ],
                      Array [
                        "L",
                        156.6,
                        82,
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
                Object {
                  "attrs": Object {
                    "cx": 130,
                    "cy": 89,
                    "fill": "#fdb05e",
                    "rx": 26.599999999999998,
                    "ry": 12,
                    "stroke": "#3b4044",
                    "x": 130,
                    "y": 82,
                  },
                  "type": "ellipse",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {},
            "mark": Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "fill": "#fdb05e",
                    "path": Array [
                      Array [
                        "M",
                        220,
                        89,
                      ],
                      Array [
                        "m",
                        -26.599999999999998,
                        0,
                      ],
                      Array [
                        "l",
                        26.599999999999998,
                        19,
                      ],
                      Array [
                        "l",
                        26.599999999999998,
                        -19,
                      ],
                      Array [
                        "l",
                        -26.599999999999998,
                        -19,
                      ],
                      Array [
                        "Z",
                      ],
                    ],
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "C",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-C",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist match example snapshot 1`] = `
Object {
  "mark": Object {
    "attrs": Object {},
    "children": Array [
      Object {
        "attrs": Object {
          "fill": null,
          "lineDash": Array [
            2,
            2,
          ],
          "lineWidth": 2,
          "stroke": "#fdb05e",
        },
        "type": "rect",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "User",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-User",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Pintora",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-Pintora",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [
          Object {
            "attrs": Object {
              "anchor": "start",
              "fill": "#f8f8f2",
              "rx": 0,
              "ry": 0,
              "stroke": "#3b4044",
            },
            "class": "activation",
            "type": "rect",
          },
        ],
        "class": "activations",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-0-0",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "render this",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [
              Object {
                "attrs": Object {
                  "fill": "#3b4044",
                  "r": 8,
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              Object {
                "attrs": Object {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "1",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-0-0",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "lineDash": Array [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-3-1",
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "Has input changed?",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [
              Object {
                "attrs": Object {
                  "fill": "#3b4044",
                  "r": 8,
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              Object {
                "attrs": Object {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "2",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-3-1",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "stroke": "#fdb05e",
            },
            "class": "loop__label-box",
            "type": "polygon",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "loop",
              "textBaseline": "top",
            },
            "class": "label-text",
            "type": "text",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "[Check input]",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "loop__title",
            "type": "text",
          },
        ],
        "class": "loop",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "lineDash": Array [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-5-1",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "your figure here",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          Object {
            "attrs": Object {},
            "children": Array [
              Object {
                "attrs": Object {
                  "fill": "#3b4044",
                  "r": 8,
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              Object {
                "attrs": Object {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "3",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-5-1",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "anchor": "start",
              "fill": "#F8EA75",
              "rx": 0,
              "ry": 0,
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "note over",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "anchor": "start",
              "fill": "#F8EA75",
              "rx": 0,
              "ry": 0,
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "note aside actor",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "anchor": "start",
              "fill": "#F8EA75",
              "rx": 0,
              "ry": 0,
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "multiline note
  is possible",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "type": "line",
          },
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#f8f8f2",
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 600,
              "text": "Divider",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "divider",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "User",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-User",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Pintora",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-Pintora",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist will render boxes 1`] = `
Object {
  "mark": Object {
    "attrs": Object {},
    "children": Array [
      Object {
        "attrs": Object {},
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#d6d3fa",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": "bold",
              "text": "group participants",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "type": "text",
          },
        ],
        "class": "sequence__participant-boxes",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Alice",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      Object {
        "attrs": Object {},
        "children": Array [],
        "class": "activations",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Alice",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      Object {
        "children": Array [
          Object {
            "attrs": Object {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          Object {
            "attrs": Object {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;
